File: SEMESTRE 3/PROG3/ClaseInicial_Chiofalo/src/ParteB/Articulo.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package ParteB;

import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author Emi Chiófalo
 */
public class Articulo {
    
    private int codigo; 
    private String denominacion; 
    private double precio; 
    private String unidadMedida; 
    private List <DetalleFactura> detalleFactura;
    
    public Articulo(int codigo, String denominacion, double precio, String unidadMedida, List<DetalleFactura> detalleFactura) {
        this.codigo = codigo;
        this.denominacion = denominacion;
        this.precio = precio;
        this.unidadMedida = unidadMedida;
        this.detalleFactura = detalleFactura;
    }

    public Articulo() {
    }

    public int getCodigo() {
        return codigo;
    }

    public void setCodigo(int codigo) {
        this.codigo = codigo;
    }

    public String getDenominacion() {
        return denominacion;
    }

    public void setDenominacion(String denominacion) {
        this.denominacion = denominacion;
    }

    public double getPrecio() {
        return precio;
    }

    public void setPrecio(double precio) {
        this.precio = precio;
    }

    public String getUnidadMedida() {
        return unidadMedida;
    }

    public void setUnidadMedida(String unidadMedida) {
        this.unidadMedida = unidadMedida;
    }

    public List<DetalleFactura> getDetalleFactura() {
        return detalleFactura;
    }

    public void addDetalleFactura(DetalleFactura detalleFactura) {
        if (this.detalleFactura == null) {
            this.detalleFactura = new ArrayList(); 
        }
        this.detalleFactura.add(detalleFactura);
    }
        
}

File: SEMESTRE 3/LAB3/TP7-TS-ModeloB/Articulo.ts
import { DetalleFactura } from "./DetalleFactura";

export class Articulo {
    constructor(codigo: number, denominacion: string, 
        precio: number, unidadMedida: string) {
        this._codigo = codigo;
        this._denominacion = denominacion;
        this._precio = precio;
        this._unidadMedida = unidadMedida;
    }

    private _codigo: number; 
    public get codigo(): number {
        return this._codigo;
    }
    public set codigo(value: number) {
        this._codigo = value;
    }

    private _denominacion: string; 
    public get denominacion(): string {
        return this._denominacion;
    }
    public set denominacion(value: string) {
        this._denominacion = value;
    }

    private _precio: number; 
    public get precio(): number {
        return this._precio;
    }
    public set precio(value: number) {
        this._precio = value;
    }

    private _unidadMedida: string; 
    public get unidadMedida(): string {
        return this._unidadMedida;
    }
    public set unidadMedida(value: string) {
        this._unidadMedida = value;
    }

    private _detalles: Array<DetalleFactura>;
    public get detalles(): Array<DetalleFactura> {
        return this._detalles;
    }
    public set detalles(value: Array<DetalleFactura>) {
        this._detalles = value;
    }
    public addDetalle(value: DetalleFactura){
        if (this._detalles == null) {
            this._detalles = new Array<DetalleFactura>();
        }
        this._detalles.push(value);
    }

}